x/mongo/driver/session/client_session.go

```diff
@@ -262,6 +274,30 @@ func (c *Client) UpdateRecoveryToken(response bson.Raw) {
 	c.RecoveryToken = token.Document()
 }
 
+// UpdateSnapshotTime updates the session's value for the atClusterTime field of ReadConcern.
+func (c *Client) UpdateSnapshotTime(response bsoncore.Document) {
+	if c == nil {
```

Comment:
Based on the spec, I think this should limit parsing on only the commands `find` / `aggregate` / `distinct`: 

> For snapshot reads the driver MUST first obtain atClusterTime from the server response of find/aggregate/distinct command

Though OTOH sending the snapshot read concern should apply to all commands.

Reply:
Done.